### ISSUE TITLE
Add Options to Postgres Connector?

### ISSUE THREAD

**Issue by tshauck** [evidence-dev/evidence#1358]
Hi, this is sorta a question and potentially a connector update request. See below for the answers to the GH issue template, but in short, I'm wondering if it's currently possible to pass postgres connection options?

I'm using the postgres wire protocol, but not an actual postgres server, so it's helpful for configuration to be able to pass options at connection time. E.g. what I do w/ `psycopg2` in python:

```python
conn = psycopg2.connect(
    host=HOST,
    port=5432,
    user=EMAIL,
    password=PASSWORD,
    dbname=DBNAME,
    options="organization=public",
)
```

So basically having the `options` parameter available.

I also quickly tried a URL encoded connection string, but that didn't work. If it's indeed possible, and I'm just dumb, I'll have another look :).

Being able to issue commands at connection-time may also work (e.g. `SET organization ...`).

Thanks!

## What do you use this database for?

Postgres, which is already supported

## Are there any popular and well-maintained clients for this database?

`pg`

## Are you interested in contributing this database connector?

Maybe, if it's easy :)

**Comment by ItsMeBrianD** [comment]
So we'll need to add the new setting to the `options` object that is exported by the postgres connector;
https://github.com/evidence-dev/evidence/blob/next/packages/postgres/index.cjs#L269-L277

This informs the UI of it's configuration options, you can make sure that the field appears properly [in your local project's settings page](http://localhost:3000/settings).

Once that's appearing properly in the UI, it should just be a matter of adding it to the [`credentials`](https://github.com/evidence-dev/evidence/blob/next/packages/postgres/index.cjs#L159-L167) that are passed to the postgres library

This is all on the `next` branch, so you'll want to branch off of that, and PR back to that.

**Comment by tshauck** [comment]
Thanks -- I tried to test my changes locally as described in [CONTRIBUTING.md](https://github.com/evidence-dev/evidence/blob/next/CONTRIBUTING.md#contribute-code), but am running into:

```
pnpm run dev:example-project

> @ dev:example-project /Users/thauck/personal/code/github.com/tshauck/evidence
> pnpm --filter ./sites/example-project dev


> @evidence-dev/components@3.0.0-usql.27 dev /Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project
> vite dev --port 3000



  VITE v4.3.9  ready in 878 ms

  ➜  Local:   http://127.0.0.1:3000/
  ➜  Network: use --host to expose
  ➜  press h to show help
/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project/src/pages/+layout.js:24
                throw new Error('Unable to load source manifest. Do you need to run build:sources?');
                      ^

Error: Unable to load source manifest. Do you need to run build:sources?
    at Module.loadDB (/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project/src/pages/+layout.js:24:9)
/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @evidence-dev/components@3.0.0-usql.27 dev: `vite dev --port 3000`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
```

Trying what it suggests leads to:

```
$ pnpm run build:sources
 ERR_PNPM_NO_SCRIPT  Missing script: build:sources

Command "build:sources" not found. Did you mean "pnpm run build:docs"?
```

**Comment by archiewood** [comment]
hmm, I think CONTRIBUTING is not yet updated on the `next` branch.

we probably should add a convenience script for this but to build:sources you probably need to

```
pnpm i 
cd sites/example-project
pnpm run build:sources
cd ../../
pnpm run dev:core-components
```

and in another terminal
```
pnpm run dev:example-project
```

**Comment by tshauck** [comment]
Thanks, that got me unblocked. There's some assumptions (e.g. big  query creds), but I think I'll have something to review after the weekend.

**Comment by ItsMeBrianD** [comment]
> There's some assumptions (e.g. big query creds), but I think I'll have something to review after the weekend.

If you run with `pnpm run build:sources --sources psql`, then that will prevent the bigquery issues from appearing
